File: frontend/src/api/design.ts
import apiClient from './index'

const api = apiClient

export interface DesignTask {
  id: number
  title: string
  description: string | null
  task_type: string
  status: string
  priority: string
  assignee_id: number | null
  assignee_name: string | null
  due_date: string | null
  community_id: number
  created_by_user_id: number | null
  content_id: number | null
  content_title: string | null
  created_at: string
  updated_at: string
}

export interface DesignTaskListItem {
  id: number
  title: string
  task_type: string
  status: string
  priority: string
  assignee_id: number | null
  assignee_name: string | null
  due_date: string | null
  content_id: number | null
  content_title: string | null
  created_at: string
  updated_at: string
}

export interface PaginatedDesignTasks {
  items: DesignTaskListItem[]
  total: number
  page: number
  page_size: number
}

export interface DesignTaskCreate {
  title: string
  description?: string | null
  task_type?: string
  priority?: string
  assignee_id?: number | null
  due_date?: string | null
  content_id?: number | null
}

export interface DesignTaskUpdate {
  title?: string
  description?: string | null
  task_type?: string
  status?: string
  priority?: string
  assignee_id?: number | null
  due_date?: string | null
  content_id?: number | null
}

export const TASK_TYPE_LABELS: Record<string, string> = {
  poster: '海报',
  icon: '图标',
  illustration: '插图',
  logo: 'Logo',
  template: '模板',
  brand_guide: '品牌规范',
  other: '其他',
}

export const TASK_STATUS_LABELS: Record<string, string> = {
  not_started: '未开始',
  in_progress: '进行中',
  review: '待审核',
  completed: '已完成',
}

export const TASK_PRIORITY_LABELS: Record<string, string> = {
  low: '低',
  medium: '中',
  high: '高',
}

export const listDesignTasks = (params?: {
  status?: string
  task_type?: string
  priority?: string
  assignee_id?: number
  page?: number
  page_size?: number
}) => api.get<PaginatedDesignTasks>('/design-tasks/', { params })

export const createDesignTask = (data: DesignTaskCreate) =>
  api.post<DesignTask>('/design-tasks/', data)

export const getDesignTask = (id: number) =>
  api.get<DesignTask>(`/design-tasks/${id}`)

export const updateDesignTask = (id: number, data: DesignTaskUpdate) =>
  api.put<DesignTask>(`/design-tasks/${id}`, data)

export const updateDesignTaskStatus = (id: number, status: string) =>
  api.patch<DesignTask>(`/design-tasks/${id}/status`, { status })

export const deleteDesignTask = (id: number) =>
  api.delete(`/design-tasks/${id}`)

File: backend/app/schemas/design.py
from datetime import date, datetime

from pydantic import BaseModel

# ─── Design Task ─────────────────────────────────────────────────────────────


class DesignTaskCreate(BaseModel):
    title: str
    description: str | None = None
    task_type: str = "other"
    priority: str = "medium"
    assignee_id: int | None = None
    due_date: date | None = None
    content_id: int | None = None


class DesignTaskUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    task_type: str | None = None
    status: str | None = None
    priority: str | None = None
    assignee_id: int | None = None
    due_date: date | None = None
    content_id: int | None = None


class DesignTaskStatusUpdate(BaseModel):
    status: str


class DesignTaskOut(BaseModel):
    id: int
    title: str
    description: str | None
    task_type: str
    status: str
    priority: str
    assignee_id: int | None
    assignee_name: str | None  # 经办人姓名（冗余字段，方便前端展示）
    due_date: date | None
    community_id: int
    created_by_user_id: int | None
    content_id: int | None
    content_title: str | None  # 关联内容标题（冗余字段）
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class DesignTaskListItem(BaseModel):
    id: int
    title: str
    task_type: str
    status: str
    priority: str
    assignee_id: int | None
    assignee_name: str | None
    due_date: date | None
    content_id: int | None
    content_title: str | None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# ─── Asset ───────────────────────────────────────────────────────────────────


class AssetUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    asset_type: str | None = None
    tags: list[str] | None = None


class AssetOut(BaseModel):
    id: int
    name: str
    description: str | None
    asset_type: str
    file_url: str
    file_size: int | None
    mime_type: str | None
    tags: list[str]
    community_id: int
    uploaded_by_user_id: int | None
    uploader_name: str | None  # 上传者姓名
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
